--- src/buildtools/third_party/libunwind/BUILD.gn	2023-07-10 10:19:16
+++ src/buildtools/third_party/libunwind/BUILD.gn	2023-07-10 10:19:23
@@ -23,7 +23,7 @@ config("libunwind_config") {
 source_set("libunwind") {
   visibility = [ "//buildtools/third_party/libc++abi" ]
   if (is_android) {
-    visibility += [ "//services/tracing/public/cpp" ]
+    visibility += [ "//services/tracing/public/cpp", "//build/config:common_deps" ]
   }
   if (!is_component_build) {
     defines = [ "_LIBUNWIND_DISABLE_VISIBILITY_ANNOTATIONS" ]
--- src/build/config/BUILD.gn	2023-07-10 10:23:49
+++ src/build/config/BUILD.gn	2023-07-10 10:23:54
@@ -246,6 +246,8 @@ group("common_deps") {
 
   if (use_custom_libcxx) {
     public_deps += [ "//buildtools/third_party/libc++" ]
+  } else {
+    public_deps += [ "//buildtools/third_party/libunwind" ]
   }
 
   if (use_afl) {
