--- usr/aarch64-linux-gnu/include/features.h	2025-10-10 10:13:09.372273427 +0800
+++ /usr/aarch64-linux-gnu/include/features.h	2025-10-10 10:23:49.987671498 +0800
@@ -261,7 +261,7 @@
 /* This is to enable the ISO C23 extension.  */
 #if (defined _ISOC23_SOURCE || defined _ISOC2Y_SOURCE \
      || (defined __STDC_VERSION__ && __STDC_VERSION__ > 201710L))
-# define __GLIBC_USE_ISOC23 	1
+# define __GLIBC_USE_ISOC23 	0
 #else
 # define __GLIBC_USE_ISOC23	0
 #endif
