diff --git a/call/payload_type_picker.cc b/call/payload_type_picker.cc
index 45105a1054..d12c3b34e2 100644
--- a/call/payload_type_picker.cc
+++ b/call/payload_type_picker.cc
@@ -370,7 +370,7 @@ void PayloadTypeRecorder::Rollback() {
 RTCError RtpHeaderExtensionRecorder::AddMapping(RtpHeaderExtensionId id,
                                                 absl::string_view uri,
                                                 bool encrypt) {
-  auto it = uri_to_id_.find(std::pair{uri, encrypt});
+  auto it = uri_to_id_.find(std::pair{std::string(uri), encrypt});
   if (it != uri_to_id_.end()) {
     if (it->second != id) {
       RTC_HISTOGRAM_BOOLEAN(
@@ -393,7 +393,7 @@ RTCError RtpHeaderExtensionRecorder::AddMapping(RtpHeaderExtensionId id,
 RTCErrorOr<RtpHeaderExtensionId> RtpHeaderExtensionRecorder::LookupId(
     absl::string_view uri,
     bool encrypt) const {
-  auto it = uri_to_id_.find(std::pair{uri, encrypt});
+  auto it = uri_to_id_.find(std::pair{std::string(uri), encrypt});
   if (it == uri_to_id_.end()) {
     return RTCError(RTCErrorType::INVALID_PARAMETER,
                     "No ID found for extension");
