diff --git a/config/c++/c++.gni b/config/c++/c++.gni
index c29d898fb..5ef014e51 100644
--- a/config/c++/c++.gni
+++ b/config/c++/c++.gni
@@ -62,7 +62,7 @@ declare_args() {
   # case.
   # We disable that on LibFuzzer builds because it breaks the libfuzzer
   # runtime. See crbug.com/411020147.
-  use_llvm_libatomic = !is_apple && !is_nacl && !use_libfuzzer
+  use_llvm_libatomic = !is_apple && !is_nacl && !use_libfuzzer && !is_linux
 }
 
 if (use_implicit_libcxx_modules) {
