diff --git a/modules/video_coding/codecs/test/android_codec_factory_helper.cc b/modules/video_coding/codecs/test/android_codec_factory_helper.cc
index f6a5046210..103d979d2a 100644
--- a/modules/video_coding/codecs/test/android_codec_factory_helper.cc
+++ b/modules/video_coding/codecs/test/android_codec_factory_helper.cc
@@ -54,9 +54,9 @@ void InitializeAndroidObjects() {
 std::unique_ptr<VideoEncoderFactory> CreateAndroidEncoderFactory() {
   JNIEnv* env = AttachCurrentThreadIfNeeded();
   ScopedJavaLocalRef<jclass> factory_class =
-      GetClass(env, "org/webrtc/HardwareVideoEncoderFactory");
+      GetClass(env, "livekit/org/webrtc/HardwareVideoEncoderFactory");
   jmethodID factory_constructor = env->GetMethodID(
-      factory_class.obj(), "<init>", "(Lorg/webrtc/EglBase$Context;ZZ)V");
+      factory_class.obj(), "<init>", "(Llivekit/org/webrtc/EglBase$Context;ZZ)V");
   ScopedJavaLocalRef<jobject> factory_object =
       ScopedJavaLocalRef<jobject>::Adopt(
           env, env->NewObject(factory_class.obj(), factory_constructor,
@@ -69,9 +69,9 @@ std::unique_ptr<VideoEncoderFactory> CreateAndroidEncoderFactory() {
 std::unique_ptr<VideoDecoderFactory> CreateAndroidDecoderFactory() {
   JNIEnv* env = AttachCurrentThreadIfNeeded();
   ScopedJavaLocalRef<jclass> factory_class =
-      GetClass(env, "org/webrtc/HardwareVideoDecoderFactory");
+      GetClass(env, "livekit/org/webrtc/HardwareVideoDecoderFactory");
   jmethodID factory_constructor = env->GetMethodID(
-      factory_class.obj(), "<init>", "(Lorg/webrtc/EglBase$Context;)V");
+      factory_class.obj(), "<init>", "(Llivekit/org/webrtc/EglBase$Context;)V");
   ScopedJavaLocalRef<jobject> factory_object =
       ScopedJavaLocalRef<jobject>::Adopt(
           env, env->NewObject(factory_class.obj(), factory_constructor,
diff --git a/sdk/android/BUILD.gn b/sdk/android/BUILD.gn
index 06fcf84a9d..627e204582 100644
--- a/sdk/android/BUILD.gn
+++ b/sdk/android/BUILD.gn
@@ -1493,11 +1493,13 @@ if (current_os == "linux" || is_android) {
   generate_jni("generated_environment_jni") {
     sources = [ "api/org/webrtc/Environment.java" ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_metrics_jni") {
     sources = [ "api/org/webrtc/Metrics.java" ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   # Generated JNI for public JNI targets, matching order of targets
@@ -1511,6 +1513,7 @@ if (current_os == "linux" || is_android) {
       "src/java/org/webrtc/JniCommon.java",
     ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_video_jni") {
@@ -1540,11 +1543,13 @@ if (current_os == "linux" || is_android) {
       "src/java/org/webrtc/WrappedNativeI420Buffer.java",
     ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_video_egl_jni") {
     sources = [ "src/java/org/webrtc/EglBase10Impl.java" ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_libvpx_vp8_jni") {
@@ -1554,6 +1559,7 @@ if (current_os == "linux" || is_android) {
     ]
 
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_libvpx_vp9_jni") {
@@ -1563,18 +1569,21 @@ if (current_os == "linux" || is_android) {
     ]
 
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_libaom_av1_encoder_jni") {
     sources = [ "api/org/webrtc/LibaomAv1Encoder.java" ]
 
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_dav1d_jni") {
     sources = [ "api/org/webrtc/Dav1dDecoder.java" ]
 
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_swcodecs_jni") {
@@ -1584,11 +1593,13 @@ if (current_os == "linux" || is_android) {
     ]
 
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_rtcerror_jni") {
     sources = [ "src/java/org/webrtc/RtcError.java" ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_peerconnection_jni") {
@@ -1603,11 +1614,13 @@ if (current_os == "linux" || is_android) {
       "src/java/org/webrtc/PlatformCertificateVerifier.java",
     ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_java_audio_jni") {
     sources = [ "api/org/webrtc/audio/JavaAudioDeviceModule.java" ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_builtin_audio_codecs_jni") {
@@ -1650,6 +1663,7 @@ if (current_os == "linux" || is_android) {
       "api/org/webrtc/BuiltinAudioEncoderFactoryFactory.java",
     ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   # Generated JNI for native API targets
@@ -1661,17 +1675,20 @@ if (current_os == "linux" || is_android) {
       "src/java/org/webrtc/WebRtcClassLoader.java",
     ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   # Generated JNI for internal targets.
 
   generate_jni("generated_logging_jni") {
     sources = [ "src/java/org/webrtc/JNILogging.java" ]
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_audio_device_module_base_jni") {
     sources = [ "src/java/org/webrtc/audio/WebRtcAudioManager.java" ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   generate_jni("generated_java_audio_device_module_native_jni") {
@@ -1680,6 +1697,7 @@ if (current_os == "linux" || is_android) {
       "src/java/org/webrtc/audio/WebRtcAudioTrack.java",
     ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 }
 
@@ -1937,6 +1955,7 @@ if (is_android) {
       "native_unittests/org/webrtc/PeerConnectionFactoryInitializationHelper.java",
     ]
     namespace = "webrtc::jni"
+    package_prefix = android_package_prefix
   }
 
   robolectric_binary("android_sdk_junit_tests") {
diff --git a/sdk/android/api/org/webrtc/PeerConnectionFactory.java b/sdk/android/api/org/webrtc/PeerConnectionFactory.java
index 8cfa52b033..e878b80f36 100644
--- a/sdk/android/api/org/webrtc/PeerConnectionFactory.java
+++ b/sdk/android/api/org/webrtc/PeerConnectionFactory.java
@@ -90,7 +90,7 @@ public class PeerConnectionFactory {
       private String fieldTrials = "";
       private boolean enableInternalTracer;
       private NativeLibraryLoader nativeLibraryLoader = new NativeLibrary.DefaultLoader();
-      private String nativeLibraryName = "jingle_peerconnection_so";
+      private String nativeLibraryName = "lkjingle_peerconnection_so";
       @Nullable private Loggable loggable;
       @Nullable private Severity loggableSeverity;
 
diff --git a/sdk/android/src/jni/jni_helpers.h b/sdk/android/src/jni/jni_helpers.h
index 950190ece0..7d7b3618c7 100644
--- a/sdk/android/src/jni/jni_helpers.h
+++ b/sdk/android/src/jni/jni_helpers.h
@@ -29,10 +29,10 @@
 // boundary. crbug.com/655248
 #define JNI_FUNCTION_DECLARATION(rettype, name, ...)                    \
   __attribute__((force_align_arg_pointer)) extern "C" JNIEXPORT rettype \
-      JNICALL Java_org_webrtc_##name(__VA_ARGS__)
+      JNICALL Java_livekit_org_webrtc_##name(__VA_ARGS__)
 #else
 #define JNI_FUNCTION_DECLARATION(rettype, name, ...) \
-  extern "C" JNIEXPORT rettype JNICALL Java_org_webrtc_##name(__VA_ARGS__)
+  extern "C" JNIEXPORT rettype JNICALL Java_livekit_org_webrtc_##name(__VA_ARGS__)
 #endif
 
 namespace webrtc {
diff --git a/sdk/android/src/jni/simulcast_video_encoder.cc b/sdk/android/src/jni/simulcast_video_encoder.cc
index 6d4ba5b3b6..48d3cb3541 100644
--- a/sdk/android/src/jni/simulcast_video_encoder.cc
+++ b/sdk/android/src/jni/simulcast_video_encoder.cc
@@ -15,7 +15,14 @@ extern "C" {
 #endif
 
 // (VideoEncoderFactory primary, VideoEncoderFactory fallback, VideoCodecInfo info)
-JNIEXPORT jlong JNICALL Java_org_webrtc_SimulcastVideoEncoder_nativeCreateEncoder(JNIEnv *env, jclass klass, jlong webrtcEnvRef, jobject primary, jobject fallback, jobject info) {
+JNI_FUNCTION_DECLARATION(jlong,
+                         SimulcastVideoEncoder_nativeCreateEncoder,
+                         JNIEnv *env,
+                         jclass klass,
+                         jlong webrtcEnvRef,
+                         jobject primary,
+                         jobject fallback,
+                         jobject info) {
     RTC_LOG(LS_INFO) << "Create simulcast video encoder";
     auto info_ref = JavaParamRef<jobject>::CreateLeaky(env, info);
     SdpVideoFormat format = VideoCodecInfoToSdpVideoFormat(env, info_ref);
diff --git a/tools_webrtc/android/build_aar.py b/tools_webrtc/android/build_aar.py
index 2fbce97c60..130b3d5fa7 100755
--- a/tools_webrtc/android/build_aar.py
+++ b/tools_webrtc/android/build_aar.py
@@ -254,7 +254,7 @@ def BuildAar(build_dir,
             Collect(aar_file, build_dir, arch, unstripped)
 
     license_dir = os.path.dirname(os.path.realpath(output_file))
-    GenerateLicenses(license_dir, build_dir, archs)
+    #GenerateLicenses(license_dir, build_dir, archs)
 
 
 def main():
diff --git a/webrtc.gni b/webrtc.gni
index 0d2db4d1d6..2468184608 100644
--- a/webrtc.gni
+++ b/webrtc.gni
@@ -234,6 +234,10 @@ declare_args() {
   #          hasn't been registered.
   rtc_strict_field_trials = ""
 
+  if (is_android) {
+    android_package_prefix = "livekit"
+  }
+
   # If different from "", symbols exported with RTC_OBJC_EXPORT will be prefixed
   # with this string.
   # See the definition of RTC_OBJC_TYPE_PREFIX in the code.
diff --git a/third_party/jni_zero/BUILD.gn b/third_party/jni_zero/BUILD.gn
index 58e0233e52..325b0e36c7 100644
--- a/third_party/jni_zero/BUILD.gn
+++ b/third_party/jni_zero/BUILD.gn
@@ -17,6 +17,7 @@ generate_jni("generate_jni") {
     ":*",
     "//components/cronet/android/*",
   ]
+  package_prefix = "livekit"
 }
 
 # Common classes to save projects from having to generate them.
diff --git a/third_party/jni_zero/codegen/header_common.py b/third_party/jni_zero/codegen/header_common.py
index 3967f94ed5..1f5cee6d81 100644
--- a/third_party/jni_zero/codegen/header_common.py
+++ b/third_party/jni_zero/codegen/header_common.py
@@ -11,7 +11,8 @@ def class_accessors(sb, java_classes):
   for java_class in java_classes:
     if java_class in (java_types.OBJECT_CLASS, java_types.STRING_CLASS):
       continue
-    escaped_name = java_class.to_cpp()
+    escaped_name = java_class.class_without_prefix.to_cpp()
+    print("header accessor: ", escaped_name)
     # #ifdef needed when multple .h files are #included that common classes.
     sb(f"""\
 #ifndef {escaped_name}_clazz_defined
@@ -42,7 +43,7 @@ def class_accessor_expression(java_class):
   if java_class == java_types.STRING_CLASS:
     return 'jni_zero::g_string_class'
 
-  return f'{java_class.to_cpp()}_clazz(env)'
+  return f'{java_class.class_without_prefix.to_cpp()}_clazz(env)'
 
 
 def header_preamble(script_name,
@@ -53,9 +54,9 @@ def header_preamble(script_name,
                     is_shared_header=False):
   if header_guard is None:
     if is_shared_header:
-      header_guard = f'{java_class.to_cpp()}_SHARED_JNI'
+      header_guard = f'{java_class.class_without_prefix.to_cpp()}_SHARED_JNI'
     else:
-      header_guard = f'{java_class.to_cpp()}_JNI'
+      header_guard = f'{java_class.class_without_prefix.to_cpp()}_JNI'
   sb = []
   sb.append(f"""\
 // This file was generated by
diff --git a/third_party/jni_zero/jni_zero.gni b/third_party/jni_zero/jni_zero.gni
index edc5364c59..40726f5cfb 100644
--- a/third_party/jni_zero/jni_zero.gni
+++ b/third_party/jni_zero/jni_zero.gni
@@ -123,7 +123,7 @@ template("_invoke_jni_zero") {
 #   namespace: Registration functions will be wrapped into this. (optional)
 #   priority_java_targets: List of java targets that, if using multiplexing,
 #     will always be placed first in the sequential switch numbers. (optional)
-#
+#   package_prefix: package prefix to add to jni
 # Example
 #   generate_jni_registration("chrome_jni_registration") {
 #     java_targets = [ ":chrome_public_apk" ]
@@ -345,6 +347,7 @@ template("generate_jni_impl") {
                                "deps",
                                "metadata",
                                "public_deps",
+                               "package_prefix",
                              ])
       if (!defined(public_deps)) {
         public_deps = []
@@ -396,6 +399,9 @@ template("generate_jni_impl") {
         if (defined(invoker.namespace)) {
           args += [ "--namespace=${invoker.namespace}" ]
         }
+        if (defined(package_prefix)) {
+          args += [ "--package-prefix=${package_prefix}" ]
+        }
         if (defined(invoker.module_name)) {
           args += [ "--module-name=${invoker.module_name}" ]
         }
