diff --git org/webrtc-sys/libwebrtc/src/modules/video_coding/codecs/h264/h264_encoder_impl.h update/webrtc-sys/libwebrtc/src/modules/video_coding/codecs/h264/h264_encoder_impl.h
index 19c16f3..af0424b 100644
--- org/webrtc-sys/libwebrtc/src/modules/video_coding/codecs/h264/h264_encoder_impl.h
+++ update/webrtc-sys/libwebrtc/src/modules/video_coding/codecs/h264/h264_encoder_impl.h
@@ -31,6 +31,7 @@
 #include "api/video_codecs/scalability_mode.h"
 #include "api/video_codecs/video_encoder.h"
 #include "common_video/h264/h264_bitstream_parser.h"
+#include "libwebrtc-hw/vpl_video_encoder.h"
 #include "modules/video_coding/codecs/h264/include/h264.h"
 #include "modules/video_coding/svc/scalable_video_controller.h"
 #include "modules/video_coding/utility/quality_scaler.h"
@@ -90,6 +91,9 @@ class H264EncoderImpl : public H264Encoder {
   }
 
  private:
+  std::unique_ptr<any_vpl::VplVideoEncoder> vplVideoEncoder_;
+  bool useHWEncoder_;
+
   SEncParamExt CreateEncoderParams(size_t i) const;
 
   webrtc::H264BitstreamParser h264_bitstream_parser_;
